pkg/engine/api/status.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+// RuleStatus represents the status of rule execution
+type RuleStatus string
+
+// RuleStatusPass is used to report the result of processing a rule.
+const (
+	// RuleStatusPass indicates that the resources meets the policy rule requirements
+	RuleStatusPass RuleStatus = "pass"
+	// RuleStatusFail indicates that the resource does not meet the policy rule requirements
+	RuleStatusFail RuleStatus = "fail"
+	// RuleStatusWarn indicates that the resource does not meet the policy rule requirements, but the policy is not scored
+	RuleStatusWarn RuleStatus = "warning"
+	// RuleStatusError indicates that the policy rule could not be evaluated due to a processing error, for
+	// example when a variable cannot be resolved  in the policy rule definition. Note that variables
+	// that cannot be resolved in preconditions are replaced with empty values to allow existence
+	// checks.
+	RuleStatusError RuleStatus = "error"
+	// RuleStatusSkip indicates that the policy rule was not selected based on user inputs or applicability, for example
+	// when preconditions are not met, or when conditional or global anchors are not satistied.
+	RuleStatusSkip RuleStatus = "skip"
+)
+
+// // String implements Stringer interface
+// func (s RuleStatus) String() string {
```

Comment:
Can we remove these?

Reply:
Yes, if CI passes... I wonder why it was so complicated, custom marshalers and so on... instead of just const strings 🤷